kafka-client/src/main/java/dev/responsive/db/CassandraFactSchema.java

```diff
@@ -76,30 +73,43 @@ public CassandraFactSchema(final CassandraClient client) {
 
   @Override
   @CheckReturnValue
-  public SimpleStatement create(final String tableName, Optional<Duration> ttl) {
+  public void create(final String tableName, Optional<Duration> ttl) {
     LOG.info("Creating fact data table {} in remote store.", tableName);
-    final CreateTable createTable = SchemaBuilder
+    CreateTableWithOptions createTable = SchemaBuilder
         .createTable(tableName)
         .ifNotExists()
-        .withPartitionKey(ROW_TYPE.column(), DataTypes.TINYINT)
         .withPartitionKey(DATA_KEY.column(), DataTypes.BLOB)
-        .withColumn(DATA_VALUE.column(), DataTypes.BLOB)
-        .withColumn(OFFSET.column(), DataTypes.BIGINT);
+        .withColumn(DATA_VALUE.column(), DataTypes.BLOB);
 
     if (ttl.isPresent()) {
       final int ttlSeconds = Math.toIntExact(ttl.get().getSeconds());
       // 20 is a magic number recommended by scylla for the number of buckets
       final Duration compactionWindow = Duration.ofSeconds(ttlSeconds / 20);
-      return createTable.withDefaultTimeToLiveSeconds(ttlSeconds)
+      createTable = createTable.withDefaultTimeToLiveSeconds(ttlSeconds)
           .withCompaction(
               SchemaBuilder.timeWindowCompactionStrategy()
                   .withCompactionWindow(
                       compactionWindow.toMinutes(),
                       TimeWindowCompactionStrategy.CompactionWindowUnit.MINUTES)
-          ).build();
+          );
     } else {
-      return createTable.withCompaction(SchemaBuilder.timeWindowCompactionStrategy()).build();
+      createTable = createTable.withCompaction(SchemaBuilder.timeWindowCompactionStrategy());
     }
+
+    // separate metadata from the main table for the fact schema, this is acceptable
+    // because we don't use the metadata at all for fencing operations and writes to
+    // it do not need to be atomic (transactional with the original table). we cannot
+    // effectively use the same table (as we do with the normal KeyValueSchema) because
+    // TWCS cannot properly compact files if there are any overwrites, which there are
+    // for the metadata columns
+    final CreateTableWithOptions createMetadataTable = SchemaBuilder
+        .createTable(metadataTable(tableName))
+        .ifNotExists()
+        .withPartitionKey(PARTITION_KEY.column(), DataTypes.INT)
```

Comment:
Are we going from a TINYINT to an INT here? Is that intentional?

Reply:
yes, TINYINT was for the "row type", the partition key on the other hand is the Kafka Partition which is technically an integer type.

Reply:
👍